--- expected
+++ baseline-and-candidate
@@ -50,8 +50,8 @@
 INFORMATION_SCHEMA 19
 U1 1
 U2 1
-DBA 549
-INFORMATION_SCHEMA 300
+DBA 548
+INFORMATION_SCHEMA 299
 U1 2
 U2 2
 ===================================================
@@ -76,8 +76,8 @@
 DBA 25
 INFORMATION_SCHEMA 19
 U1 1
-DBA 271
-INFORMATION_SCHEMA 300
+DBA 270
+INFORMATION_SCHEMA 299
 U1 2
 ===================================================
     
@@ -103,8 +103,8 @@
 INFORMATION_SCHEMA 19
 U1 1
 U2 1
-DBA 549
-INFORMATION_SCHEMA 300
+DBA 548
+INFORMATION_SCHEMA 299
 U1 2
 U2 2
 ===================================================
