diff --git a/sql/_13_issues/_14_1h/answers/bug_bts_10516.answer b/sql/_13_issues/_14_1h/answers/bug_bts_10516.answer
index be9034343..618ec90c8 100644
--- a/sql/_13_issues/_14_1h/answers/bug_bts_10516.answer
+++ b/sql/_13_issues/_14_1h/answers/bug_bts_10516.answer
@@ -11,7 +11,7 @@
 ===================================================
 0
 ===================================================
-Error:-1381
+Error:-1382
 ===================================================
 1
 ===================================================
diff --git a/sql/_15_fbo/_02_qa_test/answers/fbo_ddl02.answer b/sql/_15_fbo/_02_qa_test/answers/fbo_ddl02.answer
index ad5fe3441..5a426124c 100644
--- a/sql/_15_fbo/_02_qa_test/answers/fbo_ddl02.answer
+++ b/sql/_15_fbo/_02_qa_test/answers/fbo_ddl02.answer
@@ -9,7 +9,7 @@
 ===================================================
 0
 ===================================================
-Error:-1381
+Error:-1382
 ===================================================
 1
 ===================================================
