diff --git a/src/executables/util_service.c b/src/executables/util_service.c
index 05ced14a9..98e2a83ba 100644
--- a/src/executables/util_service.c
+++ b/src/executables/util_service.c
@@ -186,6 +186,7 @@ static UTIL_SERVICE_OPTION_MAP_T us_Service_map[] = {
   {ADMIN, UTIL_OPTION_ESTIMATE_INDEX, MASK_ADMIN},
   {ADMIN, UTIL_OPTION_LOADDB, MASK_ADMIN},
   {ADMIN, UTIL_OPTION_UNLOADDB, MASK_ADMIN},
+  {ADMIN, UTIL_OPTION_UPGRADEDB, MASK_ADMIN},
   {ADMIN, UTIL_OPTION_COMPACTDB, MASK_ADMIN},
   {ADMIN, UTIL_OPTION_PARAMDUMP, MASK_ADMIN},
   {ADMIN, UTIL_OPTION_STATDUMP, MASK_ADMIN},
