--- expected
+++ baseline-and-candidate
@@ -461,10 +461,10 @@
                           cost:  ? card ?
                inner: iscan
                           class: c node[?]
-                          index: i_a term[?] AND term[?] AND term[?]
-                          sargs: term[?]
-                          cost:  ? card ?
-               sargs: term[?]
+                          index: i_c term[?]
+                          sargs: term[?] AND term[?] AND term[?]
+                          cost:  ? card ?
+               sargs: term[?] AND term[?] AND term[?]
                cost:  ? card ?
     inner: sscan
                class: b node[?]
@@ -479,7 +479,7 @@
   HASH JOIN (inner join)
     NESTED LOOPS (inner join)
       TABLE SCAN (a)
-      INDEX SCAN (c.i_a) (key range: a.ca=c.cakey range: a.cb=c.cbkey range: a.cc=c.cc)
+      INDEX SCAN (c.i_c) (key range: a.cd=c.cd)
     TABLE SCAN (b)
 
   rewritten query: select /*+ LEADING(a) USE_NL(c) USE_HASH(a, b) NO_PARALLEL_SUBQUERY */ count(*) from [dba.ta] a, [dba.tb] b, [dba.tc] c where a.ca=b.ca and a.cb=b.cb and a.cc=b.cc and a.cd=b.cd and a.ca=c.ca and a.cb=c.cb and a.cc=c.cc and a.cd=c.cd
@@ -495,7 +495,7 @@
         SUBQUERY (uncorrelated)
           SELECT (time: ?, fetch: ?, fetch_time: ?, ioread: ?)
             SCAN (table: dba.ta), (heap time: ?, fetch: ?, ioread: ?, readrows: ?, rows: ?)
-              SCAN (index: dba.tc.i_a), (btree time: ?, fetch: ?, ioread: ?, readkeys: ?, filteredkeys: ?, rows: ?) (lookup time: ?, rows: ?)
+              SCAN (index: dba.tc.i_c), (btree time: ?, fetch: ?, ioread: ?, readkeys: ?, filteredkeys: ?, rows: ?) (lookup time: ?, rows: ?)
           SELECT (time: ?, fetch: ?, fetch_time: ?, ioread: ?)
             SCAN (table: dba.tb), (heap time: ?, fetch: ?, ioread: ?, readrows: ?, rows: ?)
      
