diff --git a/win/cci_version.h b/win/cci_version.h
index 1b91c6d..45a637e 100644
--- a/win/cci_version.h
+++ b/win/cci_version.h
@@ -22,8 +22,8 @@
 #define CCI_MAJOR_VERSION 11
 #define CCI_MINOR_VERSION 3
 #define CCI_PATCH_VERSION 0
-#define CCI_EXTRA_VERSION 0047
+#define CCI_EXTRA_VERSION 0049
 
-#define CCI_BUILD_NUMBER 11.3.0.0047-0c828ee
+#define CCI_BUILD_NUMBER 11.3.0.0049-79d0888
 
 #endif /* _VERSION_CCI_H_ */
