--- expected
+++ baseline-and-candidate
@@ -168,16 +168,16 @@
 7010005     xuebaocai     ancient mansion of jia     10000000006     7010005     5     30.0     5     chemistry     
 
 Query plan:
-nl-join (cross join)
-    outer: idx-join (inner join)
+idx-join (inner join)
+    outer: nl-join (cross join)
                outer: iscan
                           class: t? node[?]
                           index: t?_idx term[?]
-                          sargs: term[?]
                           cost:  ? card ?
                inner: iscan
                           class: t? node[?]
                           index: t?_idx term[?]
+                          sargs: term[?]
                           cost:  ? card ?
                cost:  ? card ?
     inner: iscan
